src/common/utils.lua

```diff
@@ -208,7 +268,20 @@ function utils.createHandler(tagName, tagValue, handler, position)
 		utils.camelCase(tagValue),
 		Handlers.utils.continue(Handlers.utils.hasMatchingTag(tagName, tagValue)),
 		function(msg)
-			-- sometimes the message id is not present on dryrun
+			-- handling for eth EIP-55 format, returns address if is not eth address
+			msg.From = utils.formatAddress(msg.From)
+			local knownAddressTags = {
+				"Recipient",
+				"Controller",
+			}
+			for _, tName in ipairs(knownAddressTags) do
+				-- Format all incoming addresses
+				msg.Tags[tName] = msg.Tags[tName] and utils.formatAddress(msg.Tags[tName]) or nil
+				-- aos assigns tag values to the base message level as well
```

Comment:
interesting. so theres not really any need to explicitly reach into `Tags` object?